nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -78,6 +85,80 @@ let
     }
   '';
 
+  getPoolFilesystems = pool:
+    filter (x: x.fsType == "zfs" && (fsToPool x) == pool) config.system.build.fileSystems;
+
+  getPoolMounts = prefix: pool:
+    let
+      mountPoint = fs: escapeSystemdPath (prefix + fs.mountPoint);
+    in
+      map (x: "${mountPoint x}.mount") (getPoolFilesystems pool);
+
+  createImportService = { pool, systemd, force, prefix ? "" }:
+    nameValuePair "zfs-import-${pool}" {
+      description = "Import ZFS pool \"${pool}\"";
+      # we need systemd-udev-settle until https://github.com/zfsonlinux/zfs/pull/4943 is merged
```

Comment:
The linked PR is closed, might be good to update this comment.

Reply:
Yea. Should probably leave the link, since it would be good for someone to actually finish that PR, but its status should be reflected.